src/circuit/ops/poly.rs

```diff
@@ -49,7 +49,7 @@ pub enum PolyOp<F: PrimeField + TensorType + PartialOrd> {
     Pack(u32, u32),
     GlobalSumPool,
     Iff,
-    RangeCheck(i32),
+    RangeCheck(Tolerance),
```

Comment:
This is probably more suitable as a `HybridOp` as it leverage both lookups + regular constraints. See the general review for a description as to why that may be a better fit :)